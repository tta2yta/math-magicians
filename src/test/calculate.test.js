import calculate from "../logic/calculate"
import Calculator from "../components/calculator"
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';


describe(" Operation of Calulator", ()=>{
  test('should return object with null property values when "AC" button is clicked', () => {
    const obj= {total:null, next: 10, operation:"AC"}
    expect(calculate(obj, obj.operation)).toMatchObject({next: null, operation: null, total: null});
  });
  
  it('should return correct object reperesentation when "+" button is clicked', ()=>{
    const obj= {total: 5, next: 8, operation:"+"}
    expect(calculate(obj, obj.operation)). toMatchObject({next: null, operation: "+", total: "13"})
  });

  it('should return correct object reperesentation when "-" button is clicked', ()=>{
    const obj= {total: 5, next: 8, operation:"-"}
    expect(calculate(obj, obj.operation)). toMatchObject({next: null, operation: "-", total: "-3"})
  });
  it('should return correct object reperesentation when "x" button is clicked', ()=>{
    const obj= {total: 6, next: 9, operation:"x"}
    expect(calculate(obj, obj.operation)). toMatchObject({next: null, operation: "x", total: "54"})
  });
  it('should return correct object reperesentation when "รท" button is clicked', ()=>{
    const obj= {total: 9, next: 3, operation:"รท"}
    expect(calculate(obj, obj.operation)). toMatchObject({next: null, operation: "รท", total: "3"})
  });
  it('should return correct object reperesentation when "%" button is clicked', ()=>{
    const obj= {total: 4, next: 3, operation:"%"}
    expect(calculate(obj, obj.operation)). toMatchObject({next: null, operation: "%", total: "1"})
  });

  it('should return correct object reperesentation when "=" button is clicked', ()=>{
    const obj= {total: 4, next: 3, operation:"+"}
    expect(calculate(obj, obj.operation)). toMatchObject({next:null, operation: "+", total: "7"})
  });

})


// beforeEach(() => {
//   const component = render(<Calculator />);
//   getByTestId = component.getByTestId;
//   console.log(getByTestId)
// });

// it('renders the Calculator component', () => {
//   const component = renderer.create(<Calculator />).toJSON();

//   expect(component).toMatchSnapshot();
// });

// test('should give correct subtraction operation', () => {
//   const resultHeading = getByTestId('mainResult');
//   const subtractBtn = getByTestId('-');
//   const equalsBtn = getByTestId('=');
//   const numberOnebtn = getByTestId('1');
//   const numberTwobtn = getByTestId('2');
//   console.log(equalsBtn)

//   expect(subtractBtn.textContent).toBe('-');

//   resultHeading.textContent=2
//   fireEvent.click(numberOnebtn);
//   // fireEvent.click(subtractBtn);
//   // fireEvent.click(numberTwobtn);
//   // fireEvent.click(equalsBtn);
//   // expect(resultHeading.textContent).toBe('-1');
// });