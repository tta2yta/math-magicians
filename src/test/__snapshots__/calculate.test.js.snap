// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Calculator component 1`] = `
<div
  className="calculator"
>
  <div>
    <div
      className="calculator-body"
    >
      <div
        className="input-val"
      >
        <input
          data-testid="mainResult"
          onChange={[Function]}
          type="text"
          value={0}
        />
      </div>
    </div>
    <div
      className="calculator-body"
    >
      <div
        className="calc-row"
        data-testid="AC"
        onClick={[Function]}
      >
        AC
      </div>
      <div
        className="calc-row"
        data-testid="+/-"
        onClick={[Function]}
      >
        +/-
      </div>
      <div
        className="calc-row"
        data-testid="%"
        onClick={[Function]}
      >
        %
      </div>
      <div
        className="calc-row arthemetic"
        data-testid=""
        onClick={[Function]}
      >
        รท
      </div>
    </div>
    <div
      className="calculator-body"
    >
      <div
        className="calc-row"
        data-testid="7"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="calc-row"
        data-testid="8"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="calc-row"
        data-testid="9"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="calc-row arthemetic"
        data-testid="x"
        onClick={[Function]}
      >
        x
      </div>
    </div>
    <div
      className="calculator-body"
    >
      <div
        className="calc-row"
        data-testid="4"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="calc-row"
        data-testid="5"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="calc-row"
        data-testid="6"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="calc-row arthemetic"
        data-testid="-"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="calculator-body"
    >
      <div
        className="calc-row"
        data-testid="1"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="calc-row"
        data-testid="2"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="calc-row"
        data-testid="3"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="calc-row  arthemetic"
        data-testid="+"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="calculator-body-last"
    >
      <div
        className="calc-row cols-span-2"
        data-testid="0"
        onClick={[Function]}
      >
        0
      </div>
      <div
        className="calc-row"
        data-testid="."
        onClick={[Function]}
      >
        .
      </div>
      <div
        className="calc-row arthemetic"
        data-testid="="
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
  <div
    className="math-logic"
  >
    <h2>
      Let's do some math
    </h2>
  </div>
</div>
`;
